src/Features/Core/Portable/IntroduceUsingStatement/AbstractIntroduceUsingStatementCodeRefactoringProvider.cs

```diff
@@ -230,41 +232,98 @@ private static (SyntaxTriviaList sameLine, SyntaxTriviaList endOfLine) SplitTrai
             ISyntaxFactsService syntaxFactsService,
             CancellationToken cancellationToken)
         {
-            foreach (var nodeOrToken in declarationSyntax.Parent.ChildNodesAndTokens().Reverse())
+            var firstVariableUsage = new Dictionary<ILocalSymbol, int>();
+            var lastVariableUsage = new Dictionary<ILocalSymbol, int>();
+            var localVariables = new List<ILocalSymbol> { localVariable };
+
+            // Take all the statements after the trigger variable's declaration.
+            var statementsAfterDeclaration = declarationSyntax.Parent.ChildNodesAndTokens()
+                .Select(nodeOrToken => nodeOrToken.AsNode())
+                .OfType<TStatementSyntax>()
+                .SkipWhile(node => node != declarationSyntax)
+                .ToArray();
+
+            for (var statementIndex = 0; statementIndex < statementsAfterDeclaration.Length; statementIndex++)
             {
-                var node = (TStatementSyntax)nodeOrToken.AsNode();
-                if (node is null)
+                var node = statementsAfterDeclaration[statementIndex];
+                if (ContainsReference(node, localVariables, semanticModel, syntaxFactsService, cancellationToken, out var referencedVariable))
                 {
-                    continue;
+                    // Update the statement index this variable was last used in.
+                    lastVariableUsage[referencedVariable] = statementIndex;
                 }
 
-                if (node == declarationSyntax)
+                if (node is TLocalDeclarationSyntax localDeclarationSyntax)
```

Comment:
you can add declarations with more than just TLocalDeclStatements.  For example, you might have an expr statement that has an `out-var`.  

Consider you are using IOp, you might want to just collect all the IOps for local vars in your container and select the ones declared after you.